sympy/polys/agca/modules.py

```diff
@@ -357,7 +357,7 @@ def convert(self, elem, M=None):
             if len(tpl) != self.rank:
                 raise CoercionFailed
             return FreeModuleElement(self, tpl)
-        elif elem is 0:
+        elif isinstance(elem, int) and not isinstance(elem, bool) and elem == 0:
```

Comment:
```suggestion
        elif elem is not False and elem == 0 and isinstance(elem, int):
```

Reply:
@smichr Can you have a look at this way, ``` type(elem) is type(0) and elem == 0 ```?

Reply:
The disadvantage of using `type(x) is y` is that it doesn't work for instances of subclasses of y. There are a number of different integer classes around including SymPy's Integer which in fact is not even a subclass of int. SymPy even has a different class for zero itself...

Maybe the best way to do this is using as_int from sympy.core.compatiblity. Then:
```python
try:
    elem = as_int(elem)
except ValueError:
    # Not an int
else:
    if elem == 0:
        # Do stuff
```

Reply:
We don't need it to work for subclasses, right?. It only needs to work with a pure int, which is 0, which is what the original code intended with ``` elem is 0 ```

Reply:
I think my edit suggestion is the most direct way to affirm that you have an int equal to 0.

Reply:
Alright, I'll add that in. Are there any other changes I should make?

Reply:
(`as_int` would pass with S.Zero, for example, but S.Zero is 1 would fail.)

Reply:
> disadvantage of using `type(x) is y` is that it doesn't work for instances of subclasses

It took me a second read to see that this is what we want here. Polys is pretty strict about its input. If it was looking for `foo is 1` it meant a literal Python 1. Using `if _aresame(elem, 1):` would be equivalent to checking type and value but would be more transparent than doing a test for not being False.

Reply:
So should I just import it from sympy.core.basic use that instead?

Reply:
IMO, that is the most transparent way to say what you are doing.